diff --git a/vendor/civicrm/civicrm-core/CRM/Core/BAO/Setting.php b/vendor/civicrm/civicrm-core/CRM/Core/BAO/Setting.php
index 66f425bdf..6eaa5374d 100644
--- a/CRM/Core/BAO/Setting.php
+++ b/CRM/Core/BAO/Setting.php
@@ -562,11 +562,7 @@ public static function onChangeEnvironmentSetting($oldValue, $newValue, $metadat
 
   public static function loggingMetadataCallback(array &$setting): void {
     // Disable field on setting form if system does not meet requirements
-    if (\CRM_Core_I18n::isMultilingual()) {
-      $setting['description'] = ts('Logging is not supported in multilingual environments.');
-      $setting['html_attributes']['disabled'] = 'disabled';
-    }
-    elseif (!\CRM_Core_DAO::checkTriggerViewPermission(FALSE)) {
+    if (!\CRM_Core_DAO::checkTriggerViewPermission(FALSE)) {
       $setting['description'] = ts("In order to use this functionality, the installation's database user must have privileges to create triggers (in MySQL 5.0 – and in MySQL 5.1 if binary logging is enabled – this means the SUPER privilege). This install either does not seem to have the required privilege enabled.");
       $setting['html_attributes']['disabled'] = 'disabled';
     }
