diff --git a/vendor/civicrm/civicrm-core/CRM/Core/Config.php b/vendor/civicrm/civicrm-core/CRM/Core/Config.php
index 5ff852ec1..5498d7aaf 100644
--- a/vendor/civicrm/civicrm-core/CRM/Core/Config.php
+++ b/vendor/civicrm/civicrm-core/CRM/Core/Config.php
@@ -88,12 +88,12 @@ public static function &singleton($loadFromDB = TRUE, $force = FALSE) {
       \Civi\Core\Container::boot($loadFromDB);
 
       if ($loadFromDB && self::$_singleton->dsn) {
-        Civi::service('settings_manager')->bootComplete();
 
         self::$_singleton->userSystem->postContainerBoot();
 
         $domain = \CRM_Core_BAO_Domain::getDomain();
         \CRM_Core_BAO_ConfigSetting::applyLocale(\Civi::settings($domain->id), $domain->locales);
+        Civi::service('settings_manager')->bootComplete();
 
         unset($errorScope);
 
