diff --git a/vendor/civicrm/civicrm-core/Civi/Api4/Query/Api4Query.php b/vendor/civicrm/civicrm-core/Civi/Api4/Query/Api4Query.php
index 2e014afc5..aa84d346e 100644
--- a/vendor/civicrm/civicrm-core/Civi/Api4/Query/Api4Query.php
+++ b/vendor/civicrm/civicrm-core/Civi/Api4/Query/Api4Query.php
@@ -371,7 +371,7 @@ public function composeClause(array $clause, string $type, int $depth) {
         }
         elseif ($exprA->getType() === 'SqlFunction') {
           // If $valueA uses a date extraction function and $valueB uses a relative date, add that function to $valueB, to compare apples with apples
-          if ($exprA->getCategory() === SqlFunction::CATEGORY_PARTIAL_DATE && is_string($valueB) && $valueB !== '') {
+          if ($exprA->getCategory() === SqlFunction::CATEGORY_PARTIAL_DATE && is_string($valueB) && $valueB !== '' && !str_contains($valueB, '%')) {
             $valueB = FormattingUtil::formatDateValue('YmdHis', $valueB, $operator);
             // If the formatter didn't convert it to an array, add the function, otherwise no need as we're using BETWEEN
             if (is_string($valueB)) {
